### ISSUE TITLE
Bug: NS_ERROR_MALFORMED_URI when clicking "Try it out!"

### ISSUE THREAD

**Issue by redbar0n** [Moove-it/swaggard#25]
I'm getting an NS_ERROR_MALFORMED_URI error when clicking "Try it out!" on an API action.

I think it might have something to do with the URL then attempted being set to:
localhost:3000/swagger#!/api%2Fattachments/read_only_token
with %2F being a part of the url after the url parsing, when it should probably be / instead.

I'm using Firefox 45.7.0

Chrome Version 55.0.2883.95 (64-bit) gave this error:

```
Uncaught DOMException: Failed to execute 'open' on 'XMLHttpRequest': Invalid URL
    at Request.end (http://localhost:3000/assets/swaggard/swagger-ui-6d108ff7325bed701fc9c9c3caeb6dc3.js?body=1:20914:9)
    at SuperagentHttpClient.execute (http://localhost:3000/assets/swaggard/swagger-ui-6d108ff7325bed701fc9c9c3caeb6dc3.js?body=1:4435:5)
    at module.exports.SwaggerHttp.execute (http://localhost:3000/assets/swaggard/swagger-ui-6d108ff7325bed701fc9c9c3caeb6dc3.js?body=1:4258:12)
    at module.exports.Operation.execute (http://localhost:3000/assets/swaggard/swagger-ui-6d108ff7325bed701fc9c9c3caeb6dc3.js?body=1:7548:30)
    at C.n.submitOperation (http://localhost:3000/assets/swaggard/swagger-ui-6d108ff7325bed701fc9c9c3caeb6dc3.js?body=1:23311:27)
    at HTMLInputElement.r (http://localhost:3000/assets/swaggard/lib/lodash.min-748e65e6387ba21a89990d57e17ef20c.js?body=1:1:10605)
    at HTMLLIElement.dispatch (http://localhost:3000/assets/swaggard/lib/jquery-1.8.0.min-ff0dd4be3e0f3362ad5970d13fe9ccad.js?body=1:2:5616)
    at HTMLLIElement.s (http://localhost:3000/assets/swaggard/lib/jquery-1.8.0.min-ff0dd4be3e0f3362ad5970d13fe9ccad.js?body=1:2:1585)
Request.end @ client.js:812
SuperagentHttpClient.execute @ http.js:292
SwaggerHttp.execute @ http.js:115
Operation.execute @ operation.js:891
submitOperation @ OperationView.js:455
r @ lodash.min-748e65e….js?body=1:1
dispatch @ jquery-1.8.0.min-ff0dd4b….js?body=1:2
s @ jquery-1.8.0.min-ff0dd4b….js?body=1:2
```

**Comment by redbar0n** [comment]
Tried downgrading swaggard as far back as to v0.2.0, as I thought it might be due to a bug in the latest swagger-ui which swaggard referenced, but the problem still persisted.

**Comment by redbar0n** [comment]
I troubleshooted and figured out that the error comes from the swagger JSON that swaggard generates. I copied the JSON output and tried running it through the combo `swagger_yard-rails` and `swagger-ui_rails2`, and those gave the same NS_ERROR_MALFORMED_URI error with that JSON, although they work fine with the JSON they generate themselves. 

I want to use swaggard though, to just have one gem, and since it renders descriptions correctly, and authentication headers, which the other option doesn't.

**Comment by adrian-gomez** [comment]
Hi, can you build an small example app? or at least give some code I can try to replicate locally?

Thanks!

**Comment by redbar0n** [comment]
I found the problem. It was because I was using the example the README suggested for `config.api_base_path = 'http://localhost:3000/api'`

While this is incorrect according to the swagger spec:
http://swagger.io/specification/
```
basePath
string
The base path on which the API is served, which is relative to the host.
If it is not included, the API is served directly under the host. 
The value MUST start with a leading slash (/). 
The basePath does not support path templating.
```

So changing to this, fixed the problem:
`config.api_base_path = '/api'`

**Comment by redbar0n** [comment]
Fixed it and submitted a pull request: https://github.com/adrian-gomez/swaggard/pull/26

**Comment by redbar0n** [comment]
It should be noted, and perhaps added to the README (I'll leave that decision up to you), that setting:
```ruby
config.api_base_path = '/api'
```

While also using a namespace for the API on the server:
```ruby
Rails.application.routes.draw do
  namespace :api do
    # your api resources / routes
    resources :users
  end
end
```

Will make the request url come out like this, with double api subpaths (which is probably not the intention):
```
http://localhost:3000/api/api/users/
```

So in case one has an api namespace (which is best practise?), you'd have to use:
```ruby
config.api_base_path = '/'
```
Or simply delete that line entirely, as swaggard will use '/' as default, in that case.

**Comment by adrian-gomez** [comment]
fixed by #26
